src/libsyntax/parse/parser.rs

```diff
@@ -6161,4 +6189,15 @@ impl<'a> Parser<'a> {
             _ =>  Err(self.fatal("expected string literal"))
         }
     }
+
+    fn span_to_snippet(&self, span: Span) -> String {
+        match self.sess.codemap().span_to_snippet(span) {
+            Ok(s) => s.to_owned(),
+            Err(_) => String::new(),
+        }
+    }
+
+    fn is_doc_comment(&self, s: String) -> bool {
```

Comment:
Ah right sorry, forgot to comment about this, but perhaps this function could take a `Span` directly? The function above may be a bit confusing if called in other contexts because errors are translated to empty strings (which may not be expected). Here the error case could be handled directly by just returning `false` which makes more sense in the context of querying whether a span is a doc comment or not.